pkg/mqtt5/client.go

```diff
@@ -0,0 +1,144 @@
+package mqtt5
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+
+	"net"
+	"strconv"
+
+	"github.com/IOTechSystems/go-mod-edge-utils/pkg/bootstrap/interfaces"
+	"github.com/IOTechSystems/go-mod-edge-utils/pkg/log"
+	"github.com/IOTechSystems/go-mod-edge-utils/pkg/mqtt5/config"
+	"github.com/eclipse/paho.golang/paho"
+)
+
+const (
+	AuthModeNone             = "none"
+	AuthModeUsernamePassword = "usernamepassword"
+	AuthModeCert             = "clientcert"
+	AuthModeCA               = "cacert"
+
+	SecretUsernameKey = "username"
+	SecretPasswordKey = "password"
+	SecretClientKey   = "clientkey"
+	SecretClientCert  = AuthModeCert
+	SecretCACert      = AuthModeCA
+)
+
+type Mqtt5Client struct {
+	configuration config.Mqtt5Config
+	authData      SecretData
+	mqtt5Client   *paho.Client
+	connect       *paho.Connect
+	isConnected   bool
+}
+
+type SecretData struct {
+	Username     string
+	Password     string
+	KeyPemBlock  string
+	CertPemBlock string
+	CaPemBlock   string
+}
+
+func NewMqtt5Client(config config.Mqtt5Config) Mqtt5Client {
+	return Mqtt5Client{
+		configuration: config,
+		mqtt5Client: paho.NewClient(paho.ClientConfig{
+			ClientID: config.ClientID,
+		}),
+		connect: &paho.Connect{
+			ClientID:   config.ClientID,
+			KeepAlive:  config.KeepAlive,
+			CleanStart: config.CleanStart,
+		},
+	}
+}
+func (c *Mqtt5Client) SetAuthData(secretProvider interfaces.SecretProvider, logger log.Logger) error {
+	logger.Infof("Setting auth data for secure MessageBus with AuthMode='%s' and SecretName='%s",
+		c.configuration.AuthMode,
+		c.configuration.SecretName)
+	if secretProvider == nil {
+		logger.Error("No secret provider")
+		return nil
+	}
+	secrets, err := secretProvider.GetSecret(c.configuration.SecretName)
+	if err != nil {
+		return fmt.Errorf("Unable to get Secret Data for secure message bus: %w", err)
+	}
+
+	switch c.configuration.AuthMode {
```

Comment:
use the authMode variable I added to ignore the case
`authMode := strings.ToLower(c.configuration.AuthMode)`